### DISCUSSION TITLE
导出 md + assets 格式

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#4041] [插件说明]
描述
---

将当页面转换为 Markdown 并将其中的图片保存到本地，适合 Obsidian、Typora 用户使用。

地址
---

[https://simpread.ksria.cn/plugins/details/BvjrJA6eh5](https://simpread.ksria.cn/plugins/details/BvjrJA6eh5)

视频
---

https://user-images.githubusercontent.com/81074/173170371-d7915fbb-4843-43eb-a1ab-51ba39c87f42.mp4

前提
---

此功能依赖于 **同步助手 1.3.0 版**  [下载地址](https://simpread.pro/docs/#/Sync?id=下载)

Obsidian 用户
---

如果你是 Obsidian 用户，且在使用 [导入到 Obsidian 插件](https://github.com/Kenshin/simpread/discussions/2902) 此功能也内置到了此插件中，可直接在此插件中使用 [详细说明](https://github.com/Kenshin/simpread/discussions/2902#discussioncomment-10447881)

使用
---

支持 **手动导出** 或 **加入自动化时导出** [详细说明](https://github.com/Kenshin/simpread/discussions/4041#discussioncomment-2921855)

选项
---

随便进入一个页面 e.g. https://sspai.com/post/71576 并进入 **阅读模式 → 右下角 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

<img width="1283" alt="image" src="https://github.com/user-attachments/assets/8910d05e-02b5-44dc-b297-779505e605e7" />

使用前提
---

此功能依赖于同步助手，所以需要确保

1.  在使用此功能前，需要已经安装并 [绑定了同步助手](https://simpread.pro/docs/#/Sync?id=%E9%AA%8C%E8%AF%81)

2. 确保你的**同步助手是 1.3.0**

设置目录
---

<details><summary><h4>需要设置导入的目录后才能使用，支持两种设置方案</h4></summary>

1️⃣ 同步助手增强导出 **（首选方案）**

<img width="883" height="642" alt="image" src="https://github.com/user-attachments/assets/51b008fb-9746-4888-9766-98fcf408c7bf" />

你可以直接填入下面现成的内容

```
{"extension":"assets", "path":"/xxx/Obsidian/SimpRead"}
```

注意：上述语法来自 [同步助手的增强导出](https://simpread.pro/docs/#/Sync?id=%E5%A2%9E%E5%BC%BA%E5%AF%BC%E5%87%BA) Mac 用户可以直接将 `/xxx/Obsidian/SimpRead` 替换为自己的路径

Windows 用户需要使用 `\\` 方案

```
{"extension":"assets", "path":"C:\\Users\\***\\Documents\\***\\Desktop\\Documents"}
```

2️⃣ 导出 md + assets 格式设置

<img width="608" height="223" alt="image" src="https://github.com/user-attachments/assets/52fe0237-a35d-4ca3-a2b0-6e24710b3532" />

Mac 用户的路径 `/xxx/Obsidian/SimpRead` 

Windows 用户路径 `C:\\Users\\***\\Documents\\***\\Desktop\\Documents`

但更建议使用 1️⃣ 方案，结合 `type` 语法规则，可以实现更灵活的导入方案

</details>

功能
---

1. 支持 Metadate [详细说明](https://github.com/Kenshin/simpread/discussions/4041#discussioncomment-2920472)
   
    <img width="695" height="329" alt="image" src="https://github.com/user-attachments/assets/c0b68d66-4135-4805-86b2-5c79c44dda78" />
   
2. 导出位置

    <img width="575" height="163" alt="image" src="https://github.com/user-attachments/assets/4976abe1-ccff-4e0d-94c9-56af22aa587f" />

3. [以 idx-title 的形式导出](https://github.com/Kenshin/simpread/discussions/4041#discussioncomment-2926944)

     <img width="578" height="156" alt="image" src="https://github.com/user-attachments/assets/25d381ac-cd09-453d-9591-d896d8ea482c" />
   
4. [图片导出位置](https://github.com/Kenshin/simpread/discussions/4041#discussioncomment-3920696)

     <img width="595" height="667" alt="image" src="https://github.com/user-attachments/assets/c73620a4-7aa5-4e6f-b944-a76d4379b370" />

5. [图片导出类型](https://github.com/Kenshin/simpread/discussions/4041#discussioncomment-3920120)

     <img width="595" height="163" alt="image" src="https://github.com/user-attachments/assets/cd63575c-1675-443a-a347-2c5e1ef2a573" />

快捷键
---

`a t`

关联
---

- [利用 Hazel 将 textbundle 修改为 Obsidian 可以预览的文件](https://github.com/Kenshin/simpread/discussions/3241)

- [巧用 Textbundle 格式让 Obsidian 更好的支持「包含图片子文件夹的 Markdown」](https://github.com/Kenshin/simpread/discussions/2360)

- [利用图床插件内置的 Cloudinary 图床解决个别网站无法转换图片的问题（进而解决无法使用 assets + md 方案或 Textbundle 方案的问题）](https://github.com/Kenshin/simpread/discussions/6598)

- [使用简悦提供的【离线下载辅助增强插件】与【图床插件】解决在导出时图片无法显示的问题](https://github.com/Kenshin/simpread/discussions/6597)

安装与升级
---

- 没有【使用同步助手 · 自动同步】的话，可通过 [此链接](https://simpread.ksria.cn/plugins/details/BvjrJA6eh5) 安装。

- 已经【使用同步助手 · 自动同步】的话，请使用 [插件管理器](https://www.yuque.com/kenshin/simpread/fcw60u#ymS1C) 安装。

更新日志
---

### 1.1.3

- `Optimize` 优化转换为 base64 的算法，用于解决 [[sspai.com](http://sspai.com/)](http://sspai.com) 无法转换的问题

- `Optimize` 优化设置界面到 2.0

### 1.1.2

- `Add` 新的字符，主要针对双链笔记用户去掉了一些特别字符，包括：`[` `]` `#` [#5035](https://github.com/Kenshin/simpread/issues/5035)

### 1.1.1

- `Fix` 修复非高级账户权限时导出异常问题。

### 1.1.0

此功能是 [同步助手 1.1.3](https://github.com/Kenshin/simpread/discussions/4049#discussioncomment-4192563) 增加的新功能，只能在 1.1.3 中使用。

- `Add` [图片导出位置](https://github.com/Kenshin/simpread/discussions/4041#discussioncomment-3920696) 与 Typora 保存一直的四种类型。

- `Add` [图片导出类型](https://github.com/Kenshin/simpread/discussions/4041#discussioncomment-3920120)，专门针对个别页面无法成功导出图片情况下，可使用 `Base64` 方案。

### 1.0.2

- `Fix` 当安装了 [题图](https://github.com/Kenshin/simpread/discussions/2487) 插件并且使用 [Live Editor](https://github.com/Kenshin/simpread/discussions/2852) 修改标题后，导入时仍使用旧标题的错误。

### 1.0.1

- `Fix` 当标题出现特殊字符时无法使用的错误。

### 1.0.0

- 快捷键 `a e` 支持

- [自定义 Metadata](https://github.com/Kenshin/simpread/discussions/4041#discussioncomment-2920472)

- [设定导出目录](https://github.com/Kenshin/simpread/discussions/4041#discussioncomment-2920749)

- [以 idx-title 的形式导出](https://github.com/Kenshin/simpread/discussions/4041#discussioncomment-2926944)

**Comment by Kenshin** [comment]
# Metadata 用法

Metadata 只能用于全文导出，且内置了如下占位符

| 占位符                    | 描述                      |
|------------------------|-------------------------|
| `{{url}}`                | 当前页面的 URL               |
| `{{title}}`              | 当存在稍后读时为稍后读标题，否则为原网页标题  |
| `{{org_title}}`          | 原网页标题                   |
| `{{un_title}}`           | 稍后读标题，不存在稍后读时为 untitled |
| `{{desc}}`               | 当前页面的描述                 |
| `{{cover}}`               | 当前页面的题图，没有的话用第一张图替代                 |
| `{{note}}`               | 当前稍后读的备注                |
| `{{date}}`               | 导入到 Obsidian 的时间        |
| `{{timestamp}}`          | 导入到 Obsdian 的时间戳        |
| `{{now\|yyyyddmmHHMMss}}` | 日期格式化 [详细说明](https://github.com/Kenshin/simpread/discussions/2902#discussioncomment-2590775)              |
| `{{tags}}`               | 如果当前文章是稍后读的话，为当前稍后读的标签  |
| `{{#\|tag\| }}`           | 定制标签的前/后缀 [详细说明](https://github.com/Kenshin/simpread/discussions/2902#discussioncomment-2590798)          |

**Reply by Kohlenstoff1999** [reply]
在引用{{title}}前后有换行和空格
e.g.
**模板：**
title:: {{date}}-{{title}}
url:: {{url}}
**Logseq中效果：**
title:: 2023-11-01 12:12:36-
            
高手的合成研发逻辑，10 分推荐！
          
url:: https://mp.weixin.qq.com/s/4lfi-OA6HH2na0kJLAK8_w
**询问一下如何变成以下效果？**
title:: 2023-11-01 12:12:36-高手的合成研发逻辑，10 分推荐！
url:: https://mp.weixin.qq.com/s/4lfi-OA6HH2na0kJLAK8_w

**Reply by Kenshin** [reply]
暂时没有太好的方案，使用 [Live Editor](https://github.com/Kenshin/simpread/discussions/2852) 手动去除标题的多余空行（把标题全部删除，然后再重新编写）

**Comment by Kenshin** [comment]
# md + assets 格式

为什么要使用 `md + assets` 格式以及使用它们的优势是什么，简单的做了下介绍。

## Markdown 的「问题」

Markdown 包含的是在线图片，当图片失效时就无法显示，所以简悦引入了 `离线 Markdown` 方案。

## 离线 Markdown

Markdown 可以包含离线图片，也就是将图片转为 Base64 编码形式，存储在 Markdown 中以防止图片 404 的方案，如下图所示

<img width="696" height="313" alt="image" src="https://github.com/user-attachments/assets/baf75a17-368c-4f3e-85f6-5c25843237d2" />

虽然避免了图片 404 问题，但因为 `.md` 过大会导致大部分 Markdown 都没法正常使用，我测试过：vs code、Obsidian、Typora 目前能较好的预览这种方式的只有 vs code。

## [Textbundle](http://textbundle.org/)

**同步助手 1.0.1** 中引入了导出 `.textbundle` 格式，大部分 Markdown 都支持这种格式，如：Bear、FANotes（下图所示）

<img width="773" height="503" alt="image" src="https://github.com/user-attachments/assets/08d35b1d-3ba0-4431-9899-24db7382af77" />

但更多的 Markdown 编辑器，如：Obsidian、Typora 均不支持此方式。

##  md + assets 格式

Obsidian、Typora 虽然不支持 `.textbundle` 格式，但它们支持 `md + assets` 格式，如下图所示

<img width="1519" height="526" alt="image" src="https://github.com/user-attachments/assets/c6b75dbb-13d8-49d3-bdaa-3220372b37f9" />

所以 **同步助手 1.0.2** 中引入了 `md + assets` 格式。

## 关联

- [利用图床插件内置的 Cloudinary 图床解决个别网站无法转换图片的问题（进而解决无法使用 assets + md 方案或 Textbundle 方案的问题）](https://github.com/Kenshin/simpread/discussions/6598)

- [使用简悦提供的【离线下载辅助增强插件】与【图床插件】解决在导出时图片无法显示的问题](https://github.com/Kenshin/simpread/discussions/6597)

**Reply by CWenYx** [reply]
md + assets 格式现在不能导出网页了，点击没有反应。不知道怎么回事。

**Comment by Kenshin** [comment]
# 导出根目录

<img width="588" height="170" alt="image" src="https://github.com/user-attachments/assets/06ee4512-9018-4ef5-936e-bcd8e3eb2cba" />

设置导出根目录：

Mac 用户的路径 `/xxx/Obsidian/SimpRead` 

Windows 用户路径 `C:\\Users\\***\\Documents\\***\\Desktop\\Documents`

注意：Windows 用户需要使用 `\\` 方案

## 最佳方案

但更建议在 **在同步助手端设置增强导出的路径** [详细说明](https://github.com/Kenshin/simpread/discussions/2958#discussioncomment-2713824)

增强导出的路径设置如下：

```
{"extension":"assets", "path":"/Users/***/Documents/Obsidian/SimpRead"}
```

Windows 用户需要使用 `\\` 方案

```
{"extension":"assets", "path":"C:\\Users\\***\\Documents\\***\\Desktop\\Documents"}
```

## 注意

1. **同步助手端设置增强导出的路径** 的优先级大于插件端，也就是说同时设定路径的话，以 **同步助手端设置增强导出的路径** 为主。

2. 此路径为绝对地址路径。

**Comment by Kenshin** [comment]
# 使用

## 手动方式

进入阅读模式，右下角触发器 → 动作 → 插件触发器 → 导出 md + assets 格式

<img width="337" height="145" alt="image" src="https://github.com/user-attachments/assets/8a1d15fd-79bd-4404-87d0-f83beb2e5f26" />

## 自动方式

当加入稍后读后自动导出，需要安装 [自动化辅助插件](https://github.com/Kenshin/simpread/discussions/3596) 后打开此插件，找到下图对应位置，并输入 <kbd>a t</kbd> 

<img width="601" height="165" alt="image" src="https://github.com/user-attachments/assets/e5085d6f-65aa-4b0a-8d32-8675d137a40d" />

### 视频

https://user-images.githubusercontent.com/81074/173170371-d7915fbb-4843-43eb-a1ab-51ba39c87f42.mp4

**Comment by Kenshin** [comment]
# 以 `idx-title` 的形式导出

当用户配置了 [知识库](https://www.yuque.com/kenshin/simpread/wkswh7) 方案时，**建议开启此功能**，这样当加入稍后读后导出时，会以  `idx-title` 为标题。

<img width="560" height="138" alt="image" src="https://github.com/user-attachments/assets/5b84c9bc-2c73-4f4a-82af-c0731999f0ab" />

导出后的标题

<img width="402" height="166" alt="image" src="https://github.com/user-attachments/assets/fd153125-58c8-48f5-b09e-21d3c7744404" />

**Comment by Kenshin** [comment]
# 图片的导出类型

<img width="577" height="145" alt="image" src="https://github.com/user-attachments/assets/6fbab8c8-e2fa-42c5-84e8-fc0e974c3198" />

支持 `HTTP` 与 `Base64` ，前者为默认值。

##  `HTTP` 方案

将当前阅读模式的图片直接发送到同步助手中，在同步助手中保存到本地。

##   `Base64`  方案

将当前阅读模式的图片先转换为 base64 格式，然后直接通过同步助手保存到本地。

一般用于：某些网页有一些限制导致无法使用 `HTTP` 保存图片，当使用后会出现下图所示的提示。

https://user-images.githubusercontent.com/81074/202834601-ee1271ed-aa76-4125-8050-442d00bd2d9f.mp4

**注意**

不建议使用此方式，因为「前端转换图片」会比较慢，只有在无法使用  `HTTP` 方案时再使用。

# 关联

- https://github.com/Kenshin/simpread/issues/4510

- https://github.com/Kenshin/simpread/issues/4620

**Comment by Kenshin** [comment]
# 图片导出位置

图片的导出位置在结构上与 Typora 保持一致，包含四种类型：

<img width="577" height="307" alt="image" src="https://github.com/user-attachments/assets/2931cbd1-ecb8-41c2-ae91-519a35bfc27a" />

## 1️⃣ `/${filename}/assets`

<img width="472" height="225" alt="image" src="https://github.com/user-attachments/assets/45c7d143-e090-41dc-a39f-4d54027ca481" />

## 2️⃣ `./${filename}.assets` → 保存到 `标题.assets` 文件夹下

<img width="395" height="99" alt="image" src="https://github.com/user-attachments/assets/2de55645-72f8-4225-920c-5134d3008c43" />

## 3️⃣ `./assets` → 无论是多少个文章，将图片统一全部都保持到 `assets` 文件夹下面

<img width="342" height="100" alt="image" src="https://github.com/user-attachments/assets/7b4f2d16-3a35-40bb-8e8f-0a23d3813c8d" />

## 4️⃣ `custom` → 将图片保存本地的任意目录

需要在下图所示的位置设置路径，位置在：**导出 md + assets 格式选项**

<img width="603" height="232" alt="image" src="https://github.com/user-attachments/assets/f28b0871-50d5-41c0-b89d-aebd59542bf8" />
   
下面是一个例子，按上图所示，将图片保存到  `Obsidian@simpread/media` 下面

<img width="414" height="191" alt="image" src="https://github.com/user-attachments/assets/e9b16968-a941-4e4d-befd-f09a115d6f3c" />

注意：此地址为绝对路径。

## 5️⃣ `relative path`

这是 1.2.0 增加的功能，相对 4️⃣ 的绝对路径，此方案支持 **相对路径**，使用语法如下：

```
./${filename}/attachments
./assets/${filename}.assets
```

 `.` 是根目录 `${filename}` 是当前阅读模式的标题（或稍后读），假设当前的标题叫作 `简悦支持相对路径` 则上面的内容翻译为

```
./简悦支持相对路径/attachments
./assets/简悦支持相对路径.assets
```

同时，也支持下面的方式

```
./attachments
```

即：将图片放到 **根目录的 attachments 文件夹下面**。

同时也支持更复杂的方案，使用语法 `::`

```
./${filename}::./${filename}/assets
```

即：

```
./简悦支持相对路径/简悦支持相对路径.md
./简悦支持相对路径/简悦支持相对路径/assets/
```

此结构等同于 `./${filename}/assets` 或 方案一 `/${filename}/assets`

以下是复杂方案的几种用法

```
./${filename}::./assets/${filename}.assets
./${filename}::../img/${filename}
./::./assets/${filename}.assets
./::./${filename}/attachments
./::./assets/${filename}
./Simpread/::./assets/${filename}
./${filename}::./${filename}/assets
```

**Reply by Kenshin** [reply]
# 自定义位置

需要在下图所示的位置设置路径，位置在：**导出 md + assets 格式选项**

<img width="511" alt="image" src="https://user-images.githubusercontent.com/81074/202834001-2c1fda6e-dafd-41e7-b404-b453e92e0ae6.png">
   
下面是一个例子，按上图所示，将图片保存到  `Obsidian@simpread/media` 下面

<img width="362" alt="image" src="https://user-images.githubusercontent.com/81074/202834042-51980633-47ec-492e-976d-61eaf53cc1a5.png">

### 注意

此地址为绝对路径，暂时不支持相对路径。

**Reply by ihouhuan** [reply]
请问怎么自定义导出图片的位置呀，每篇md文章一个assets文件夹，且名称总是1，2，3，文件夹多了看着好烦，想图片集中放到一个文件夹，且通过时间戳重命名图片名称，有解吗？

**Reply by se7eneo** [reply]
这周啥时候发布，还有三天，持续刷新😁

**Reply by l05k** [reply]
确实已经很不错了，一开始以为可以做到类似VSCode的相对路径支持，不过也可以自己用脚本微调一下。
主要需求是：
1. 能够将所有assets文件夹收纳 （提供 ./assets/${filename} 这类方式，会比第二个 ${filename}.assets 更好一点）
2. 迁移到相对路径的上级目录 （目前我是将各种图片统一归档到 "../img/${filename}" 目录下）
3. 此外如果能在导出后通过同步助手自动化执行脚本就更好了

**Reply by l05k** [reply]
对的。脚本的话不是说内置脚本实现，而是能够在某动作后执行本地的脚本程序（提供运行语句之类的），这样可以更丰富自定义动作。
例如，可以写脚本抽取关键词来自动分类导出的文件。（现在虽然也可以用quicker之类的分两步实现，但不够自动化）

**Reply by Kenshin** [reply]
emm 我考虑下。

想要运行的话，具体脚本格式是什么？`.sh` 吗？我需要知道具体运行哪种类型的脚本。

**Reply by foreverlong** [reply]
作者您好，我想实现图片保存在./assets/${filename}.assets这个目录下面，请问如何设置呢？

**Reply by Kenshin** [reply]
@foreverlong 

暂时不支持这种形式的定制 😂

**Reply by foreverlong** [reply]
请问md+assets导出只能下载简约同步助手的桌面端才可以使用吗？

**Reply by Kenshin** [reply]
@foreverlong

是的

**Reply by niukejun** [reply]
作者，请问！导出md+assets到obsition怎么实现：图片自动上传到picgo图床，文章内的图片全部自动转化为链接。

**Reply by Kenshin** [reply]
> 作者，请问！导出md+assets到obsition怎么实现：图片自动上传到picgo图床，文章内的图片全部自动转化为链接。

都上传到你自己的图床了，就不需要使用 `assets + md` 了吧... 😂

**Reply by GentleLee** [reply]
大佬，请问下导出图片的重命名规则是什么样的呢？

**Reply by Kenshin** [reply]
@GentleLee 

日期的时间戳 e.g. `1728783454758.png`

<img width="280" alt="image" src="https://github.com/user-attachments/assets/9bf8271c-899f-45fb-b206-d1ce54c0353e">

**Reply by GentleLee** [reply]
感谢，意思就是不会重名了？

**Reply by Kenshin** [reply]
@GentleLee 

是的。

**Reply by Philosjay** [reply]
在我目前的使用场景下，此插件支持的4种图片存放位置方案似乎都没法满足我的需求，具体来说是这样：

1. 我的obsidian库使用了**默认配置**，每个笔记中的附件（包括图片）存储在与笔记文件同目录的`attachments`文件夹中，如下：
    - 笔记1
        - 笔记1.md
        - attachments
            - 图片1.jpg
            - 图片2.jpg
2. 我使用此插件将网页导出为 md+assets，导出位置的1、2、3方案都是将图片放到`assets`文件夹中，而我的obsidian的很多附件管理功能在识别笔记的附件时，都是在与笔记文件同目录的`attachments`文件夹中寻找附件，而不在assets`文件夹中，所以这3个导出方案与我的obsidian配置不兼容。
3. 如果使用方案4导出，可以用绝对路径将图片导出到同一个`attachments`文件夹中，导出多个网页后，不同网页的图片都混在同一个`attachments`文件夹中，而我需要像方案1那样将每个文档的图片存放在同目录的文件夹中，所以方案4也无法满足我的需求。

综上，我觉得如果插件能够支持自定义`附件文件夹名称`，就能以最小的代价满足我的需求，即我可以在方案1的基础上将`assets`目录名改为`attachments`就能满足需求。

不知我的使用方式和理解是否有误，求解答。

**Reply by DustyNight** [reply]
@Kenshin 

我也有与相同的使用场景，一开始还以为能够通过custom实现，但实际试了才知道不行，希望能够支持自定义文件夹的功能

**Reply by Kenshin** [reply]
@Philosjay 

抱歉，之前漏掉了这个信息。

前三种方案其实个你的方案很接近，只不过 `assets` 无法自定义。如果定义为 `attachments`  就满足你的需求了。

@DustyNight 

简悦的第四种方案支持自定义文件夹，但不支持相对自定义文件夹，我会考虑的。

**Reply by Kenshin** [reply]
嗨，各位

md + assets 插件更新到了 1.2.0 增加了自定义图片保存路径的功能，细节请看 https://github.com/Kenshin/simpread/discussions/4041#discussioncomment-3920696 里面提到的 **5️⃣ relative path**

注意：此功能需要配套同步助手 1.3.0 [下载地址](https://simpread.pro/docs/#/Sync?id=%E4%B8%8B%E8%BD%BD) & [更新日志](https://simpreadpublish.notion.site/2482effc334e4df8aa3aac7da8bb8059?pvs=25#278907a65f5880abb3b0d1c7aae9e3fd)

***

@Philosjay @DustyNight 

解决方案是设置目录为 `./${filename}/attachments`

@foreverlong

解决方案是设置目录为 `./assets/${filename}.assets`

**Reply by iggoss** [reply]
@Kenshin 
浏览器插件中，我设置为  relative path
然后自定义路径设置为  ./assets/${filename}.assets
使用md+assets导出，结构如下：

文件名 / 文件名.md
assets / 文件名 / 图片1
assets / 文件名 / 图片2

如果我想让 md 文件，直接就在根目录呢？如何配置？
比如这样：

文件名.md
assets / 文件名 / 图片1
assets / 文件名 / 图片2

因为默认情况，md文件其实都在根目录，但如上面设置后，md反倒跑到文件名的文件夹里面去了，感觉是多套了一层？或者有意如此设计？然后如何设置让md直接在根目录呢？谢谢

**Reply by Kenshin** [reply]
@iggoss 

你可以使用更复杂的方案 `::`

按照你的预期语法应该是：`./::./assets/${filename}` 得到下图所示位置

<img width="474" height="299" alt="image" src="https://github.com/user-attachments/assets/6c144fa6-7d56-44ed-95c2-370672483d29" />

**Reply by iggoss** [reply]
测试有效，感谢作者。

**Comment by Kenshin** [comment]
# 绝对路径与相对路径

## 注意

1.2.0 增加了更强大灵活的 **定制路径规则 [5️⃣ relative path](https://github.com/Kenshin/simpread/discussions/4041#discussioncomment-3920696)**

## 详细说明

<img width="581" height="189" alt="image" src="https://github.com/user-attachments/assets/4951b279-32e3-4a3a-82f8-e4f9bbb1c89e" />

此功能仅当选择 [方案 ➍ 时](https://github.com/Kenshin/simpread/discussions/4041#discussioncomment-3930228) 可用

<img width="591" height="117" alt="image" src="https://github.com/user-attachments/assets/00b06b54-e72e-42d2-9399-9badcc0c02a2" />

## 使用绝对路径（启用此功能）

在 Typora 中预览效果

<img width="1246" height="224" alt="image" src="https://github.com/user-attachments/assets/c1ecd751-e2f8-41ef-89e0-31d34c247917" />

## 使用相对路径（禁用此功能）

在 Obsidian 中预览效果

<img width="789" height="183" alt="image" src="https://github.com/user-attachments/assets/03f84bc8-d54e-4cfe-b268-3d3a39aa5256" />

**Reply by JimmyChou1729** [reply]
您好，请问可以去掉路径中的<>字符吗？或者保留<>的意义是？这种路径格式和某些插件不兼容，比如说obsidian中的image zoom。

**Reply by Kenshin** [reply]
@JimmyChou1729 

嗨，`![](<xxx>)` 是 Markdown 语法，当 Image URL 出现空格时，如果没有做特殊处理，会在 Obsidian 出现无法识别的情况（但其它一些 Markdown 编辑器没有这个问题，因为它本身就做了处理 e.g.  Typora ）

所以，你可以去掉 `<>` 并没有什么影响... 但你提到的 image zoom 不支持这种就有点... 😂

**Reply by JimmyChou1729** [reply]
哈学习了学习了！

**Reply by apparition957** [reply]
https://github.com/renmu123/obsidian-image-auto-upload-plugin
`![](<xxx>)`如果引入该语法，该插件无法识别内容，导致无法上传图床

**Reply by Kenshin** [reply]
@apparition957 

请在这个插件下提示，而不是这里。这个语法细节参见 https://github.com/Kenshin/simpread/discussions/4041#discussioncomment-5895146

**Reply by apparition957** [reply]
抱歉，我查了下markdown规范，`![]()`这种才是正确的，不需要额外引入`<>`这个符号
ref: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#images

**Reply by Kenshin** [reply]
@apparition957 

`![](<xxx>)` 其中的 `<xxx>` 是为了防止 URL 断掉（有空格）的情况而设置的，大部分 Markdown 编辑器应该都支持它 e.g. Obsidian Typora Logseq 等。

PS：你查的仅仅只是 Github Markdown 规范。

<img width="772" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/3044a6b9-2de0-4344-a647-d379bd93fab5">

**Comment by 850312766** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
